### ISSUE TITLE
Cryptic error when CSVfile column names are inconsistent to data

### ISSUE THREAD

**Issue by rainwoodman** [bccp/nbodykit#281]
I get this error when the column names I gave CSVFile is one less than the number of columns in the file, instead of an error about mismatched column names.

```
Traceback (most recent call last):
  File "analysis.py", line 29, in <module>
    addrsd(halos1)
  File "analysis.py", line 22, in addrsd
    print halos['z'].compute()
  File "/home/yfeng1/.local/lib/python2.7/site-packages/dask/base.py", line 78, in compute
    return compute(self, **kwargs)[0]
  File "/home/yfeng1/.local/lib/python2.7/site-packages/dask/base.py", line 178, in compute
    results = get(dsk, keys, **kwargs)
  File "/home/yfeng1/.local/lib/python2.7/site-packages/dask/threaded.py", line 69, in get
    **kwargs)
  File "/home/yfeng1/.local/lib/python2.7/site-packages/dask/async.py", line 502, in get_async
    raise(remote_exception(res, tb))
dask.async.KeyError: 0
```

**Comment by nickhand** [comment]
Huh, that is odd. What happens if you just use CSVFile directly (so no dask calls)? Is this an error reading from pandas.read_csv propagating down? 

Pandas can do some weird things when the names and columns mismatch. I thought it would just read the first N columns, where N is the number of names. We should probably put in a test where we peek at the file and check for mismatches

**Comment by nickhand** [comment]
fixed by d9391cf
